services/distribution/src/main/java/app/coronawarn/server/services/distribution/statistics/keyfigurecard/factory/HeaderCardFactory.java

```diff
@@ -46,14 +41,11 @@ public KeyFigureCard makeKeyFigureCard(StatisticsJsonStringObject stats) {
   }
 
   private void throwIfNullFieldsFound(StatisticsJsonStringObject stats) {
-    var nullFields = getNonNullFields(stats).stream()
-        .filter(f -> f.getSecond().isEmpty())
+    var nullFields = getRequiredFieldValues(stats).stream()
+        .filter(Optional::isEmpty)
         .collect(Collectors.toList());
     if (!nullFields.isEmpty()) {
-      var missingFields = nullFields.stream()
-          .map(Pair::getFirst)
-          .collect(Collectors.toList());
-      throw new MissingPropertyException(missingFields);
+      throw new MissingPropertyException(this.getCardId());
```

Comment:
This is how it worked in the past, the problem is linking the JSON property name with the actual validation. The previous implementation I passed a the string that represented what property was (e.g. "seven_day_r_value_1st_reported_daily"), but that quickly got out of sync from the actual properties used, and keeping it up-to-date was pretty cumbersome. The reason being properties were changed a lot, so to avoid having a code that can easily become misleading I decided to remove.  

Maybe could be a way to get the property name from the Jackson annotation, but I'm not sure.